compiler/optimizer/OMRLocalCSE.cpp

```diff
@@ -585,29 +595,28 @@ void OMR::LocalCSE::examineNode(TR::Node *node, SharedSparseBitVector &seenAvail
                  SharedSparseBitVector::Cursor sc(tmpAliases);
                  for (sc.SetToFirstOne(); sc.Valid(); sc.SetToNextOne())
                     {
-                    if (_storeNodesIndex[sc] != 0)
-                       {
-                       uint32_t index = _storeNodesIndex[sc];
-                       _storeNodes.RemoveEntry(index);
-                       _storeNodesIndex[sc] = 0;
-                       }
+                    StoreMap::iterator result = _storeMap->find(sc);
+                    if (result != _storeMap->end())
+                       _storeMap->erase(sc);
                     }
                  }
               }
            else
               {
-              CS2::TableOf<TR::Node *, TR::Allocator>::Cursor nc(_storeNodes);
-              for (nc.SetToFirst(); nc.Valid(); nc.SetToNext())
+              TR_BitVector storeMapSymRefs(comp()->trMemory()->currentStackRegion());
+              for (auto itr = _storeMap->begin(), end = _storeMap->end(); itr != end; ++itr)
+                 storeMapSymRefs.set(itr->first);
+
```

Comment:
Walking the entries of the _storeMap should not be significantly more expensive than walking the list. I was avoiding any semantic changes beyond memory lifetime so we could investigate further in a subsequent revision.